### DISCUSSION TITLE
[Feature] Automatic (re)setting of "Speaker setups"

### DISCUSSION THREAD

**Discussion by Coldblackice** [File-New-Project/EarTrumpet#918] [Ideas]
**tl;dr**

A super handy feature would be if EarTrumpet could automatically )set audio devices' speaker-setups to a particular arrangement, e.g. 7.1, 5.1, 2.1, Stereo/mono, without any manual intervention needed, besides setting of a rule.

**Background:**

An issue I've dealt with for years --

My audio comes via HDMI out from my Nvidia GPU and into a home theater receiver. Despite having a 2.1 setup (2 speakers + a sub), I tell Windows to send the full 7.1 to let my receiver do all the signal managing, not trusting Windows' handling. When I've let Windows only send out the "Stereo" signal, the resulting audio is much quieter.

Regardless, the issue is that for whatever reason, when I turn my monitors back on -- one of which is HDMI connected through my HT receiver -- Windows sometimes randomly sees it as a new device, setting its speaker-setup as "Stereo" instead of 7.1. I have to then manually re-set it in Windows audio-device playback dialog maybe once every week or two.

It seems to be a side effect of HDMI when an HDMI-connected monitor is turned off and back on. Windows sometimes thinks it's a new device. Not sure why. And I haven't found a way to make Windows set a higher count speaker setup by default.

It'd be useful if EarTrumpet was able to automatically set speaker-setups, even a per device speaker-setup if possible, but most useful of all, able to re-set the speaker setup if it sees that a device has been set back to "Stereo" or any other setup.

**Comment by riverar** [comment]
@Coldblackice Sounds like a great idea. Have you use EarTrumpet Actions yet? (in our [experimental builds](https://eartrumpet.app)) I'm curious what would be best for you -- a shortcut trigger to change speaker configuration? Or trigger based on app launch/terminate? All of the above?

**Reply by Coldblackice** [reply]
Thanks. I have been using the experimental build actually; the macro actions feature is a great addition.

"All of the above" would be the bees knees, if possible. If I had to choose one however, it would be the trigger-based automation. Windows continues to reset the speaker layout setup at random. If you Google it, you'll see this has been an issue affecting many people for a decade or so now.

If EarTrumpet was able to automatically detect that the Windows speaker setup was changed to X (like "Stereo") and then automatically change it back to Y (like *7.1"), that would be phenomenal, and would solve many people's issues that have had to deal with this over the years!
